sbt_common/internal_model/src/main/scala/uk/ac/wellcome/models/transformable/sierra/SierraRecordNumber.scala

```diff
@@ -6,7 +6,7 @@ object SierraRecordTypes extends Enumeration {
 
 sealed trait SierraRecordNumber {
   val recordNumber: String
-  val recordType: SierraRecordTypes.Value
+  val recordType: Option[SierraRecordTypes.Value]
```

Comment:
I'm not sure I like this being optional and then `withCheckDigit` potentially throwing an exception. It feels like `recordType` should not exist at all in `UntypedSierraRecordNumber`